x-pack/plugins/osquery/server/routes/fleet_wrapper/get_agents.ts

```diff
@@ -35,10 +36,22 @@ export const getAgentsRoute = (router: IRouter, osqueryContext: OsqueryAppContex
       async (context, request, response) => {
         let agents;
         try {
-          agents = await osqueryContext.service
-            .getAgentService()
-            ?.asInternalUser // @ts-expect-error update types
-            .listAgents(request.query);
+          agents = await osqueryContext.service.getAgentService()?.asInternalUser.listAgents({
+            ...(request.query as ListWithKuery & { showInactive: boolean }),
```

Comment:
Consider specifying query fields instead just spreading it (I know it was like this before, but would be nice improvement). 
Connected to some safety reasons about spreading data from request directly to search when used asInternalUser.